mvpa2/algorithms/searchlight_hyperalignment.py

```diff
@@ -0,0 +1,476 @@
+        # Voxel selection within Searchlight
+        # Usual metric of between-subject between-voxel correspondence
+        if self.featsel != 1.0:
+            # computing feature scores from the data
+            feature_scores = compute_feature_scores(ds, self.exclude_from_model)
+            if self.featsel <1.0:
+                fselector = FractionTailSelector(self.featsel, tail='upper', mode='select',sort=False)
+            else:
+                fselector = FixedNElementTailSelector(np.floor(self.featsel), tail='upper', mode='select',sort=False)
+            # XXX Artificially make the seed_index feature score high to keep it(?)
+            if self.use_same_features:
+                if len(self.exclude_from_model)>0:
+                    feature_scores = [feature_scores[ifs] for ifs in range(len(self.ndatasets)) 
+                                        if ifs not in self.exclude_from_model]
+                feature_scores = np.mean(np.asarray(feature_scores),axis=0)
+                feature_scores[seed_index] = max(feature_scores)
+                features_selected = fselector(feature_scores)
+                ds = [ sd[:, features_selected] for sd in ds]
+            else:
+                features_selected = []
+                for fs in feature_scores:
+                    fs[seed_index] = max(fs)
+                    features_selected.append(fselector(fs))
+                ds = [ sd[:, fsel] for fsel,sd in zip(features_selected, ds)]
+        # Try hyperalignment
+        try:
+            # it is crucial to retrain hyperalignment, otherwise it would simply
+            # project into the common space of a previous iteration
+            if len(self.exclude_from_model)==0:
+                self.hyperalignment.train(ds)
+            else:
+                self.hyperalignment.train([ds[i] for i in range(len(ds)) 
+                                            if i not in self.exclude_from_model])
+            mappers = self.hyperalignment(ds)
+            if mappers[0].proj.dtype is self.dtype:
+                mappers = [ m.proj for m in mappers]
+            else:
+                mappers = [ m.proj.astype(self.dtype) for m in mappers]
+            if self.featsel != 1.0:
+                # Reshape the projection matrix from selected to all features
+                mappers_full = [ np.zeros((nfeatures,nfeatures)) for im in range(len(mappers))]
+                if self.use_same_features:
+                    for mf,m in zip(mappers_full, mappers):
+                        mf[np.ix_(features_selected, features_selected)]=m
+                else:
+                    for mf,m,fsel in zip(mappers_full, mappers, features_selected):
+                        mf[np.ix_(fsel, features_selected[self.ref_ds])]=m
+                mappers = mappers_full
+        except LinAlgError:
+            print "SVD didn't converge. Try with a new reference, may be."
+            mappers = [np.eye(nfeatures, dtype='int')]*self.ndatasets
```

Comment:
I feel that it is better to raise an exception that to continue with identities!